buildSrc/src/main/kotlin/partiql.conventions.gradle.kts

```diff
@@ -40,6 +40,12 @@ dependencies {
     testImplementation(Deps.kotlinTest)
     testImplementation(Deps.kotlinTestJunit)
     testImplementation(Deps.junitParams)
+    // Custom ktlint rules are added by adding to the `dependencies` block: https://github.com/JLLeitschuh/ktlint-gradle/tree/v10.2.0?tab=readme-ov-file#configuration
+    // Currently, we only run the rules when the `custom-ktlint-rules` property is set.
+    // Once we enable the custom rules to run by default, this conditional can be removed.
+    if (hasProperty("custom-ktlint-rules")) {
```

Comment:
After the `internal` top-level and `public` top-level code issues are resolved, we can remove this property logic such that the custom ktlint rules will run by default.

```
# End state after existing code issues detected by custom ktlint rules are resolved
> ./gradlew ktlintCheck # will run standard and custom rules
```

Any detected issues in gradle and GH Actions will also fail the build.